pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -137,16 +137,17 @@ func (a *Bootstrap) Generate(dependencies asset.Parents) error {
 	platform := installConfig.Config.Platform.Name()
 	platformFilePath := fmt.Sprintf("bootstrap/%s/files", platform)
 	directory, err := data.Assets.Open(platformFilePath)
-	if directory != nil {
+	if err == nil && directory != nil {
```

Comment:
I believe that directory == nil if err != nil is an invariant, so simply checking err is more idiomatic

```
if err == nil {
    ...
```

but perhaps we should be careful not to swallow unexpected errors. Untested, but I'd expect something like:

```
  if err == nil {
    ...
  else if !os.IsNotExist(err) {
      return err
  }
```

or

```
  switch {
  case err != nil && !os.IsNotExist(err):
      return err
  case err == nil:
      ...
   }
```

Meh

Reply:
Fair enough, I think `err == nil` is reasonable. How sure are you about os.IsNotExist being the only one we want to swallow? Asset is actually an `http.FileSystem`.  It does seem to return os.IsNotExist, but I'm not sure I want to blow everything up if some platform-specific directory doesn't exist when most platforms don't even use it.

Reply:
Haven't looked carefully, but isn't this a vfsgen FS at runtime? That does return os.ErrNotExist it looks like ... wait, it looks like it's the only error it can return ... in which case just handling err == nil is fine I guess

Reply:
Ok pushed an update for that